cgi/product_image_upload.pl

```diff
@@ -129,20 +129,29 @@
 	}
 }
 
+
+my $response_ref = {
+	files => [{
+			filename => $filename . "",	# Make filename a scalar
+	}],
+};
+
 if ((not defined $code) or ($code eq '')) {
 
 	$log->warn("no code");
-	my %response = ( status => 'status not ok');
-	$response{error} = "error - missing product code";
+	$response_ref->{status} = 'status not ok';
```

Comment:
Are we missing translations?

Reply:
Those were intended for the API initially so I think we never added translations for them.

Reply:
not sure it's wise, especially people rely on parsing.

Reply:
We now do display the error message to users in the bulk image upload form, so I guess we could translate that. It's much less likely that an app would parse the error messages.

Reply:
I kept the "status not ok" as-is (it's never shown to users), but I added translations for error messages that can be shown to users (we already had some for some other image upload errors).